src/Item.cpp

```diff
@@ -35,11 +41,20 @@ void CItem::Init( CItemDef *pid )
 {
     m_id = pid;
     m_Color.SetColor(m_id->m_Color);
-    if( Util::GetRandom(1,100) < 5 )
+}
+
+void CItem::ApplyCursedStatus( int likelihood )
+{
+    if( CItem::PercentChance(likelihood) )
     {
         m_dwFlags |= ITEM_FLAG_CURSED;
         m_Color.SetColor(255,0,0,255);
-    }
+    }    
+}
+
+bool CItem::PercentChance ( int chance )
```

Comment:
no thank you, go look at all the places `1d100` occurs in code. The RNG is well-featured, delete this function

Reply:
Removed.